test/integration/shoots/logging/logging.go

```diff
@@ -0,0 +1,279 @@
+		framework.ExpectNoError(err)
+		err = create(ctx, f.ShootClient.Client(), fluentBitClusterRole)
+		framework.ExpectNoError(err)
+		err = create(ctx, f.ShootClient.Client(), fluentBitClusterRoleBinding)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Deploy the fluent-bit DaemonSet")
+		err = create(ctx, f.ShootClient.Client(), fluentBitConfMap)
+		framework.ExpectNoError(err)
+		err = create(ctx, f.ShootClient.Client(), fluentBit)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Wait until fluent-bit DaemonSet is ready")
+		err = f.WaitUntilDaemonSetIsRunning(ctx, f.ShootClient.Client(), fluentBitName, v1beta1constants.GardenNamespace)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Deploy the simulated cluster and shoot controlplane namespaces")
+		for i := 0; i < numberOfSimulatedClusters; i++ {
+			shootNamespace := getShootNamesapce(i)
+			ginkgo.By(fmt.Sprintf("Deploy namespace %s", shootNamespace.Name))
+			err := create(ctx, f.ShootClient.Client(), shootNamespace)
+			framework.ExpectNoError(err)
+			_, err = kutil.TryUpdateNamespace(ctx, f.ShootClient.Kubernetes(), retry.DefaultBackoff, gardenNamespace.ObjectMeta, func(ns *corev1.Namespace) (*corev1.Namespace, error) {
+				kutil.SetMetaDataLabel(&ns.ObjectMeta, "gardener.cloud", "shoot")
+				return ns, nil
+			})
+			framework.ExpectNoError(err)
+
+			cluster := getCluster(i)
+			ginkgo.By(fmt.Sprintf("Deploy cluster %s", cluster.Name))
+			err = create(ctx, f.ShootClient.DirectClient(), cluster)
+			framework.ExpectNoError(err)
+
+			ginkgo.By(fmt.Sprintf("Deploy the loki service in namespace %s", shootNamespace.Name))
+			lokiShootService := getLokiShootService(i)
+			err = create(ctx, f.ShootClient.Client(), lokiShootService)
+			framework.ExpectNoError(err)
+
+			ginkgo.By(fmt.Sprintf("Deploy the logger application in namespace %s", shootNamespace.Name))
+			loggerParams := struct {
+				HelmDeployNamespace string
+				LogsCount           int
+			}{
+				shootNamespace.Name,
+				logsCount,
+			}
+			err = f.RenderAndDeployTemplate(ctx, f.ShootClient, templates.LoggerAppName, loggerParams)
+			framework.ExpectNoError(err)
+		}
+
+		loggerLabels := labels.SelectorFromSet(labels.Set(map[string]string{
+			"app": logger,
+		}))
+		for i := 0; i < numberOfSimulatedClusters; i++ {
+			shootNamespace := fmt.Sprintf("%s%v", simulatesShootNamespacePrefix, i)
+			ginkgo.By(fmt.Sprintf("Wait until logger application is ready in namespace %s", shootNamespace))
+			err := f.WaitUntilDeploymentsWithLabelsIsReady(ctx, loggerLabels, shootNamespace, f.ShootClient)
+			framework.ExpectNoError(err)
+		}
+
+		ginkgo.By("Verify loki received logger application logs for all namespaces")
+		err = WaitUntilLokiReceivesLogs(ctx, 1*time.Minute, f, v1beta1constants.GardenNamespace, "pod_name", logger, logsCount*numberOfSimulatedClusters, f.ShootClient)
+		framework.ExpectNoError(err)
+
+	}, getLogsFromLokiTimeout, framework.WithCAfterTest(func(ctx context.Context) {
```

Comment:
I guess it would be more readable if the above function is not implemented in-place. It was very strange to me next argument here of something and started wondering what is happening.

Reply:
Just like the example above you gave me in CBeforeEach?